proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -103,13 +103,23 @@ public static Proctor construct(
                                 testDefinition,
                                 identifierValidator);
             } else {
-                testChooser =
-                        new StandardTestChooser(
-                                RuleEvaluator.EXPRESSION_FACTORY,
-                                functionMapper,
-                                testName,
-                                testDefinition,
-                                identifierValidator);
+                if (testDefinition.getEnableUnitlessAllocations()) {
+                    testChooser =
+                            new UnitlessTestChooser(
```

Comment:
just thinking out loud, can a test be a random type but also enable unitless allocation? (i know it does not make a lot of sense)

Reply:
It could enable it but we should not treat it as unitless because random test do not use an identifier (the unit).

Maybe we should gray out this feature on the FE if it is a random test.